SuiteRunner could not be initial by default Configuration

### TestNG Version
> 7.5

### Expected behavior
SuiteRunner could init by default Configuration

### Actual behavior
```
Caused by: java.lang.NullPointerException
	at org.testng.internal.objects.SimpleObjectDispenser.instantiateUsingDefaultConstructor(SimpleObjectDispenser.java:178)
	at org.testng.internal.objects.SimpleObjectDispenser.createInstance(SimpleObjectDispenser.java:87)
	... 46 more
```

### Is the issue reproducible on runner?
yes, here is a sample code
```
public class IssueTest {
  @Test
  public void issueTest() {
    XmlSuite suite = new XmlSuite();
    suite.setName("TestSuite");
    XmlTest test = new XmlTest(suite);
    test.setName("ChildTests");
    List<XmlClass> classes = new ArrayList<>();
    classes.add(new XmlClass("test.configuration.suites.IssueTest"));
    test.setXmlClasses(classes);

    final IConfiguration configuration = new Configuration();
    final boolean useDefaultListeners = true;

    SuiteRunner baseSuiteRunner = new SuiteRunner(configuration, suite,
        "outputDir",
        new ITestRunnerFactory() {

          @Override
          public TestRunner newTestRunner(ISuite suite, XmlTest xmlTest,
                                          Collection<IInvokedMethodListener> listeners, List<IClassListener> classListeners) {
            TestRunner runner =
                new TestRunner(configuration, suite, xmlTest,
                    false /* skipFailedInvocationCounts */,
                    listeners, classListeners);
            if (useDefaultListeners) {
              runner.addListener(new TestHTMLReporter());
              runner.addListener(new JUnitXMLReporter());
            }

            return runner;
          }

        },
        useDefaultListeners,
        Comparator.comparingInt(ITestNGMethod::getPriority));
    baseSuiteRunner.run();
  }
}
```

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans


### Test case sample
> Please, share the test case (as small as possible) which shows the issue


### Contribution guidelines

Incase you plan to raise a pull request to fix this issue, please make sure you refer our [Contributing](.github/CONTRIBUTING.md) section for detailed set of steps.

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,5 +1,6 @@
 Current
 7.6.0
+Fixed: GITHUB-2743: SuiteRunner could not be initial by default Configuration (Nan Liang)
 Fixed: GITHUB-2729: beforeConfiguration() listener method should be invoked for skipped configurations as well(Nan Liang)
 Fixed: assertEqualsNoOrder for Collection and Iterators size check was missing (Adam Kaczmarek)
 Fixed: GITHUB-2709: Testnames not working together with suites in suite (Martin Aldrin)

--- a/testng-core/src/main/java/org/testng/SuiteRunner.java
+++ b/testng-core/src/main/java/org/testng/SuiteRunner.java
@@ -18,6 +18,7 @@ import org.testng.internal.invokers.ConfigMethodArguments.Builder;
 import org.testng.internal.invokers.IInvocationStatus;
 import org.testng.internal.invokers.IInvoker;
 import org.testng.internal.invokers.InvokedMethod;
+import org.testng.internal.objects.ObjectFactoryImpl;
 import org.testng.internal.thread.ThreadUtil;
 import org.testng.reporters.JUnitXMLReporter;
 import org.testng.reporters.TestHTMLReporter;
@@ -148,6 +149,9 @@ public class SuiteRunner implements ISuite, IInvokedMethodListener {
     List<IMethodInterceptor> localMethodInterceptors =
         Optional.ofNullable(methodInterceptors).orElse(Lists.newArrayList());
     setOutputDir(outputDir);
+    if (configuration.getObjectFactory() == null) {
+      configuration.setObjectFactory(new ObjectFactoryImpl());
+    }
     if (suite.getObjectFactoryClass() == null) {
       objectFactory = configuration.getObjectFactory();
     } else {

--- a/testng-core/src/test/java/test/configuration/ConfigurationTest.java
+++ b/testng-core/src/test/java/test/configuration/ConfigurationTest.java
@@ -2,7 +2,9 @@ package test.configuration;
 
 import java.util.Arrays;
 import org.testng.Assert;
+import org.testng.TestNG;
 import org.testng.annotations.Test;
+import test.configuration.issue2743.SuiteRunnerIssueTestSample;
 import test.configuration.sample.ConfigurationTestSample;
 import test.configuration.sample.ExternalConfigurationClassSample;
 import test.configuration.sample.MethodCallOrderTestSample;
@@ -28,5 +30,13 @@ public class ConfigurationTest extends ConfigurationBaseTest {
   public void testSuite() {
     testConfiguration(SuiteTestSample.class);
     Assert.assertEquals(Arrays.asList(1, 2, 3, 4, 5), SuiteTestSample.m_order);
+  }
+
+  @Test
+  public void testSuiteRunnerWithDefaultConfiguration() {
+    TestNG testNG = create(SuiteRunnerIssueTestSample.class);
+    testNG.run();
+
+    Assert.assertEquals(testNG.getStatus(), 0);
   }
 }

--- a/testng-core/src/test/java/test/configuration/issue2743/SuiteRunnerIssueTestSample.java
+++ b/testng-core/src/test/java/test/configuration/issue2743/SuiteRunnerIssueTestSample.java
@@ -1,0 +1,71 @@
+package test.configuration.issue2743;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import org.testng.IClassListener;
+import org.testng.IInvokedMethodListener;
+import org.testng.ISuite;
+import org.testng.ITestNGMethod;
+import org.testng.ITestRunnerFactory;
+import org.testng.SuiteRunner;
+import org.testng.TestRunner;
+import org.testng.annotations.Test;
+import org.testng.internal.Configuration;
+import org.testng.internal.IConfiguration;
+import org.testng.reporters.JUnitXMLReporter;
+import org.testng.reporters.TestHTMLReporter;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+public class SuiteRunnerIssueTestSample {
+
+  @Test
+  public void suiteRunnerSample() {
+    XmlSuite suite = new XmlSuite();
+    suite.setName("TestSuite");
+    XmlTest test = new XmlTest(suite);
+    test.setName("ChildTests");
+    List<XmlClass> classes = new ArrayList<>();
+    classes.add(new XmlClass("test.configuration.issue2743.SuiteRunnerIssueTestSample"));
+    test.setXmlClasses(classes);
+
+    final IConfiguration configuration = new Configuration();
+    final boolean useDefaultListeners = true;
+
+    SuiteRunner suiteRunner =
+        new SuiteRunner(
+            configuration,
+            suite,
+            "outputDir",
+            new ITestRunnerFactory() {
+
+              @Override
+              public TestRunner newTestRunner(
+                  ISuite suite,
+                  XmlTest xmlTest,
+                  Collection<IInvokedMethodListener> listeners,
+                  List<IClassListener> classListeners) {
+                TestRunner runner =
+                    new TestRunner(
+                        configuration,
+                        suite,
+                        xmlTest,
+                        false /* skipFailedInvocationCounts */,
+                        listeners,
+                        classListeners);
+                if (useDefaultListeners) {
+                  runner.addListener(new TestHTMLReporter());
+                  runner.addListener(new JUnitXMLReporter());
+                }
+
+                return runner;
+              }
+            },
+            useDefaultListeners,
+            Comparator.comparingInt(ITestNGMethod::getPriority));
+    suiteRunner.run();
+  }
+}

